src/aiida/engine/transports.py

```diff
@@ -117,10 +117,17 @@ def do_open():
             assert transport_request.count >= 0, 'Transport request count dropped below 0!'
             # Check if there are no longer any users that want the transport
             if transport_request.count == 0:
+                # IMPORTANT: Remove from _transport_requests BEFORE closing the transport.
+                # This prevents a race condition with async transports where:
+                # 1. close() is called, which for AsyncTransport uses run_until_complete()
+                # 2. With nest_asyncio (used by plumpy), this can yield to the event loop
+                # 3. Another task might enter and get the same transport_request
+                # 4. That task tries to use the transport that's being closed -> error
```

Comment:
tried to improve the message by being more verbose, not sure, you can adapt it as you whish
```suggestion
                # 1. close() is called, which for AsyncTransport uses run_until_complete(close_async)
                # 2. With nest_asyncio (used by plumpy), this call yields back to the event loop
                # 3. The event loop schedules close_async, then continues running a other tasks - for example one that requests the transport which is scheduled to be closed
                # 4. The task now using the transport to do some operation awaits, next the close_async task closes the transport while still in use -> error
```